File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/ChangeFontButtonOnClickListener.java
package ro.danserboi.quotesformindandsoul;

import android.content.Context;
import android.view.View;
import android.widget.TextView;

public class ChangeFontButtonOnClickListener implements View.OnClickListener{
    private final Context context;
    private int fontIdx;
    private final TextView mWordsTextView;
    private final TextView mAuthorTextView;

    private final static int FONTS_NO = 12;

    public ChangeFontButtonOnClickListener(Context context, TextView mWordsTextView, TextView mAuthorTextView) {
        this.context = context;
        this.mWordsTextView = mWordsTextView;
        this.mAuthorTextView = mAuthorTextView;
    }

    @Override
    public void onClick(View view) {
        Utils.buttonAnimation(view);
        mWordsTextView.setTypeface(FontCache.get(context.getResources().getStringArray(R.array.fonts)[fontIdx]));
        mAuthorTextView.setTypeface(FontCache.get(context.getResources().getStringArray(R.array.fonts)[fontIdx]));
        fontIdx++;
        if(fontIdx == FONTS_NO) {
            fontIdx = 0;
        }
    }
}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/ChangeFontButtonOnClickListener.kt
package ro.danserboi.quotesformindandsoul

import android.content.Context
import android.view.View
import android.widget.TextView

class ChangeFontButtonOnClickListener(private val context: Context, private val mWordsTextView: TextView, private val mAuthorTextView: TextView) : View.OnClickListener {
    private var fontIdx: Int = 0
    companion object {
        private const val FONTS_NO = 12
    }

    override fun onClick(view: View) {
        Utils.buttonAnimation(view)
        mWordsTextView.typeface = FontCache[context.resources.getStringArray(R.array.fonts)[fontIdx]]
        mAuthorTextView.typeface = FontCache[context.resources.getStringArray(R.array.fonts)[fontIdx]]
        fontIdx++
        if (fontIdx == FONTS_NO) {
            fontIdx = 0
        }
    }

}